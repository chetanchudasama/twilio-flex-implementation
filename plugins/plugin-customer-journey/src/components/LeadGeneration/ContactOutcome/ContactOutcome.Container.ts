import { connect } from "react-redux";
import { Dispatch } from "redux";
import { CustomerDetailModel } from "@common/components";

import { AppState, namespace } from "../../../states";
import { ContactOutcome } from "./ContactOutcome";
import { StateToProps, DispatchToProps } from "./ContactOutcome.Props";

const mapStateToProps = (state: AppState): StateToProps => {
  const customer = Object.assign(
    new CustomerDetailModel(),
    state[namespace].customer.customer
  );
  return {
    callbackBooked: customer.callbackBooked,
    taskDecisions: state[namespace].staticItems.taskDecisions,
  };
};

const mapDispatchToProps = (dispatch: Dispatch<any>): DispatchToProps => ({});

export default connect(mapStateToProps, mapDispatchToProps)(ContactOutcome);
