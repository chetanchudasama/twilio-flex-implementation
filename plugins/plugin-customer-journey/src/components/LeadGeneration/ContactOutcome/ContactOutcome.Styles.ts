import styled from "react-emotion";
import { borderStyle, secondary, white } from "@common/components";
import { Menu } from "@material-ui/core";

export const ContactOutcomeStyles = styled("div")({
  ".dropdown-icon-btn": {
    padding: "6px 10px",
    borderRadius: 4,
    border: borderStyle,
    color: secondary,
    minHeight: 40,
    background: white,
  },
  ".no-contact-btn": {
    padding: "6px 10px",
    borderRadius: 4,
    border: borderStyle,
    color: secondary,
    minHeight: 40,
    background: white,
    fontSize: 14,
    fontWeight: 600,
    lineHeight: "18px",
  },
});

export const MenuStyles = styled(Menu)({
  ".menu-item-divider": {
    padding: 0,
    height: 2,
  },
  ".divider": {
    width: "100%",
  },
});
