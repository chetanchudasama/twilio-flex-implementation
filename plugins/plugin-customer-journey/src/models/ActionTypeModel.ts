export class ActionTypeModel {
  public id: number;

  public name: string;

  constructor() {
    this.id = -1;
    this.name = "";
  }
}
