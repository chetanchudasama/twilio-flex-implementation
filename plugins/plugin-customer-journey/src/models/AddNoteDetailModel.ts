export class AddNoteDetailModel {
  public applicationId: number;

  public content: string;

  public isImportant: boolean;

  public sendToBroker: boolean;

  constructor() {
    this.applicationId = -1;
    this.content = "";
    this.isImportant = false;
    this.sendToBroker = false;
  }
}
