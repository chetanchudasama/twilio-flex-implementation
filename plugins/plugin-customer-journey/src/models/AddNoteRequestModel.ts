import { Manager } from "@twilio/flex-ui";

export class AddNoteRequestModel {
  public comment: string;

  public applicationId: number;

  public userId: number;

  public isImportant: boolean;

  constructor() {
    const { agentId } =
      Manager.getInstance().store.getState().flex.worker.attributes;

    this.comment = "";
    this.applicationId = -1;
    this.userId = agentId || 0;
    this.isImportant = false;
  }
}
