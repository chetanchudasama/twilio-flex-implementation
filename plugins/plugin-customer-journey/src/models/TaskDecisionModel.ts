export class TaskDecisionModel {
  taskItemDecisionId: number;

  parentTaskScheduleTemplateId: number;

  childTaskScheduleTemplateId: number | null;

  taskItemTypeId: number;

  decisionName: string;

  decisionDescription: string;

  shouldKill: boolean;

  shouldRaise: boolean;

  isSuccess: boolean;

  isContact: boolean;

  isCallback: boolean;

  isAllocation: boolean;

  state: number | null;

  createsAutomaticNote: boolean;

  automaticNote: string;

  constructor() {
    this.taskItemDecisionId = -1;
    this.parentTaskScheduleTemplateId = -1;
    this.childTaskScheduleTemplateId = null;
    this.taskItemTypeId = -1;
    this.decisionName = "";
    this.decisionDescription = "";
    this.shouldKill = false;
    this.shouldRaise = false;
    this.isSuccess = false;
    this.isContact = false;
    this.isCallback = false;
    this.isAllocation = false;
    this.state = null;
    this.createsAutomaticNote = false;
    this.automaticNote = "";
  }
}
