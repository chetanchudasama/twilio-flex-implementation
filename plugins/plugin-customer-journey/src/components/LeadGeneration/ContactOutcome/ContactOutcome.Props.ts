import { TaskDecisionModel } from "../../../models/TaskDecisionModel";

export interface StateToProps {
  callbackBooked?: Date;
  taskDecisions: TaskDecisionModel[];
}

export interface DispatchToProps {}

export interface OwnProps {
  setCallbackDetail: (callbackDate: Date | null) => void;
  handleContactOutcomeClick: (
    taskDecision: TaskDecisionModel,
    callbackBooked?: Date
  ) => void;
}

export type ContactOutcomeProps = StateToProps & DispatchToProps & OwnProps;
