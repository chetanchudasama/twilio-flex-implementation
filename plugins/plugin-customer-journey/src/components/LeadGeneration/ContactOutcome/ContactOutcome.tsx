import React, { useState, useEffect } from "react";
import { Button, Divider, Grid, IconButton, MenuItem } from "@material-ui/core";
import ArrowDropDownIcon from "@material-ui/icons/ArrowDropDown";
import { CallbackDialog } from "@common/components";

import { ContactOutcomeStyles, MenuStyles } from "./ContactOutcome.Styles";
import { ContactOutcomeProps } from "./ContactOutcome.Props";
import { TaskDecisionModel } from "../../../models/TaskDecisionModel";

const BUTTON_COUNT = 2;

export const ContactOutcome: React.FC<ContactOutcomeProps> = ({
  callbackBooked,
  setCallbackDetail,
  taskDecisions,
  handleContactOutcomeClick,
}) => {
  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
  const [openCallbackDialog, setOpenCallbackDialog] = useState<boolean>(false);
  const [selectedTaskDecision, setSelectedTaskDecision] =
    useState<TaskDecisionModel>(new TaskDecisionModel());

  useEffect(() => {
    if (openCallbackDialog && callbackBooked) {
      handleContactOutcomeClick(selectedTaskDecision, callbackBooked);
    }
    setOpenCallbackDialog(false);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [callbackBooked]);

  const handleClick = (event: React.MouseEvent<HTMLElement>) => {
    setAnchorEl(event.currentTarget);
  };

  const handleMenuItemClick = (option: TaskDecisionModel) => {
    setSelectedTaskDecision(option);
    if (option.isCallback) {
      setOpenCallbackDialog(true);
    } else {
      handleContactOutcomeClick(option);
    }
    setAnchorEl(null);
  };

  const handleSetCallbackDetail = (callBackDate: Date | null) => {
    setCallbackDetail(callBackDate);
  };

  return (
    <ContactOutcomeStyles>
      {taskDecisions.length > 0 && (
        <Grid
          container
          spacing={8}
          direction="row"
          justify="flex-end"
          wrap="nowrap"
        >
          {taskDecisions
            .slice(0, BUTTON_COUNT)
            .map((taskDecision: TaskDecisionModel) => (
              <Grid item key={taskDecision.taskItemDecisionId}>
                <Button
                  variant="outlined"
                  className="no-contact-btn"
                  onClick={() => handleContactOutcomeClick(taskDecision)}
                >
                  {taskDecision.decisionName}
                </Button>
              </Grid>
            ))}
          <Grid item>
            <IconButton onClick={handleClick} className="dropdown-icon-btn">
              <ArrowDropDownIcon />
            </IconButton>
            <MenuStyles
              anchorEl={anchorEl}
              keepMounted
              open={Boolean(anchorEl)}
              getContentAnchorEl={null}
              onClose={() => {
                setAnchorEl(null);
              }}
              anchorOrigin={{
                vertical: "bottom",
                horizontal: "left",
              }}
              className="banner-menu"
            >
              {taskDecisions.slice(BUTTON_COUNT, -1).map((option) => (
                <MenuItem
                  key={option.taskItemDecisionId}
                  onClick={() => handleMenuItemClick(option)}
                >
                  {option.decisionName}
                </MenuItem>
              ))}
              <MenuItem key={-1} className="menu-item-divider">
                <Divider className="divider" />
              </MenuItem>
              <MenuItem
                key={taskDecisions[taskDecisions.length - 1].taskItemDecisionId}
                onClick={() =>
                  handleMenuItemClick(taskDecisions[taskDecisions.length - 1])
                }
              >
                {taskDecisions[taskDecisions.length - 1].decisionName}
              </MenuItem>
            </MenuStyles>
          </Grid>
        </Grid>
      )}
      {openCallbackDialog && (
        <CallbackDialog
          open={openCallbackDialog}
          noteOptional
          callbackBooked={callbackBooked}
          setCallbackDetail={handleSetCallbackDetail}
          handleCloseDialog={() => {
            setOpenCallbackDialog(false);
          }}
        />
      )}
    </ContactOutcomeStyles>
  );
};
