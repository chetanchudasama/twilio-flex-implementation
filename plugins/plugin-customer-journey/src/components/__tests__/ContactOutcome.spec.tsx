import React from "react";
import { mount } from "enzyme";
import { create } from "react-test-renderer";
import { act as Act } from "react-dom/test-utils";

import { Menu, MenuItem } from "@material-ui/core";
import { CallbackDialog } from "@common/components";
import { ContactOutcomeProps } from "../LeadGeneration/ContactOutcome/ContactOutcome.Props";
import { ContactOutcome } from "../LeadGeneration/ContactOutcome/ContactOutcome";
import { TaskDecisionModel } from "../../models/TaskDecisionModel";

jest.mock("@common/components", () => {
  const components = jest.requireActual("@common/components");
  return {
    ...components,
    CallbackDialog: jest.fn(({ children }) => {
      return <div>{children}</div>;
    }),
  };
});

jest.mock("@material-ui/core", () => {
  const materialUI = jest.requireActual("@material-ui/core");
  return {
    ...materialUI,
    Menu: jest.fn(({ children }) => children),
  };
});

const taskDecisionList: TaskDecisionModel[] = [
  {
    taskItemDecisionId: 1,
    parentTaskScheduleTemplateId: 29,
    childTaskScheduleTemplateId: null,
    taskItemTypeId: 1,
    decisionName: "No Contact (Left VM)",
    decisionDescription: "",
    shouldKill: false,
    shouldRaise: false,
    isSuccess: false,
    isContact: false,
    isCallback: false,
    isAllocation: false,
    state: null,
    createsAutomaticNote: false,
    automaticNote: "",
  },
  {
    taskItemDecisionId: 2,
    parentTaskScheduleTemplateId: 29,
    childTaskScheduleTemplateId: null,
    taskItemTypeId: 1,
    decisionName: "No Contact (No VM)",
    decisionDescription: "",
    shouldKill: false,
    shouldRaise: false,
    isSuccess: false,
    isContact: false,
    isCallback: false,
    isAllocation: false,
    state: null,
    createsAutomaticNote: false,
    automaticNote: "",
  },
  {
    taskItemDecisionId: 4,
    parentTaskScheduleTemplateId: 29,
    childTaskScheduleTemplateId: null,
    taskItemTypeId: 1,
    decisionName: "NTU - Customer Not Interested",
    decisionDescription: "",
    shouldKill: false,
    shouldRaise: false,
    isSuccess: false,
    isContact: false,
    isCallback: false,
    isAllocation: false,
    state: null,
    createsAutomaticNote: false,
    automaticNote: "",
  },
  {
    taskItemDecisionId: 5,
    parentTaskScheduleTemplateId: 29,
    childTaskScheduleTemplateId: null,
    taskItemTypeId: 1,
    decisionName: "Warm Transfer to Account Manager",
    decisionDescription: "",
    shouldKill: false,
    shouldRaise: false,
    isSuccess: false,
    isContact: false,
    isCallback: false,
    isAllocation: true,
    state: null,
    createsAutomaticNote: false,
    automaticNote: "",
  },
  {
    taskItemDecisionId: 6,
    parentTaskScheduleTemplateId: 29,
    childTaskScheduleTemplateId: null,
    taskItemTypeId: 1,
    decisionName: "Customer Hung Up During Call",
    decisionDescription: "",
    shouldKill: false,
    shouldRaise: false,
    isSuccess: false,
    isContact: false,
    isCallback: false,
    isAllocation: false,
    state: null,
    createsAutomaticNote: false,
    automaticNote: "",
  },
  {
    taskItemDecisionId: 7,
    parentTaskScheduleTemplateId: 29,
    childTaskScheduleTemplateId: null,
    taskItemTypeId: 1,
    decisionName: "Customer Hung Up During Transfer",
    decisionDescription: "",
    shouldKill: false,
    shouldRaise: false,
    isSuccess: false,
    isContact: false,
    isCallback: false,
    isAllocation: false,
    state: null,
    createsAutomaticNote: false,
    automaticNote: "",
  },
  {
    taskItemDecisionId: 8,
    parentTaskScheduleTemplateId: 29,
    childTaskScheduleTemplateId: null,
    taskItemTypeId: 1,
    decisionName: "Call Did Not Connect",
    decisionDescription: "",
    shouldKill: false,
    shouldRaise: false,
    isSuccess: false,
    isContact: false,
    isCallback: false,
    isAllocation: false,
    state: null,
    createsAutomaticNote: false,
    automaticNote: "",
  },
  {
    taskItemDecisionId: 9,
    parentTaskScheduleTemplateId: 29,
    childTaskScheduleTemplateId: null,
    taskItemTypeId: 1,
    decisionName: "Arrange a Callback",
    decisionDescription: "",
    shouldKill: false,
    shouldRaise: false,
    isSuccess: false,
    isContact: false,
    isCallback: true,
    isAllocation: false,
    state: null,
    createsAutomaticNote: false,
    automaticNote: "",
  },
];

const props: ContactOutcomeProps = {
  taskDecisions: taskDecisionList,
  callbackBooked: new Date(2021, 8, 8),
  setCallbackDetail: jest.fn(),
  handleContactOutcomeClick: jest.fn(),
};

describe("ContactOutcome", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it("renders correctly on load", () => {
    const wrapper = create(<ContactOutcome {...props} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it("should render Menu with MenuItem, when icon button is clicked", () => {
    const wrapper = mount(<ContactOutcome {...props} />);

    wrapper.find(".dropdown-icon-btn").at(0).simulate("click");
    wrapper.update();

    expect(wrapper.find(Menu).exists()).toBeTruthy();
    expect(wrapper.find(MenuItem).exists()).toBeTruthy();
    expect(wrapper.find(MenuItem).length).toBe(
      taskDecisionList.slice(2, -1).length + 2
    );
  });

  it("should emit handleContactOutcomeClick method, when No Contact (Left VM) button is clicked", () => {
    const wrapper = mount(<ContactOutcome {...props} />);

    const button = wrapper.find(".no-contact-btn");

    button.at(0).simulate("click");
    wrapper.update();

    expect(wrapper.props().handleContactOutcomeClick).toHaveBeenCalledTimes(1);
  });

  it("should emit handleContactOutcomeClick method, when Call Did Not Connect menu item is clicked", () => {
    const wrapper = mount(<ContactOutcome {...props} />);

    wrapper.find(".dropdown-icon-btn").at(0).simulate("click");
    wrapper.update();

    wrapper.find(MenuItem).at(0).simulate("click");
    wrapper.update();

    expect(wrapper.props().handleContactOutcomeClick).toHaveBeenCalledTimes(1);
  });

  it("should not render CallbackDialog, when value of isCallback is false and Arrange a Callback menu item is clicked", () => {
    props.taskDecisions[props.taskDecisions.length - 1].isCallback = false;
    const wrapper = mount(<ContactOutcome {...props} />);

    wrapper.find(".dropdown-icon-btn").at(0).simulate("click");
    wrapper.update();

    wrapper.find(MenuItem).last().simulate("click");
    wrapper.update();

    expect(wrapper.find(CallbackDialog).exists()).toBeFalsy();
  });

  it("should render CallbackDialog, when value of isCallback is true and Arrange a Callback menu item is clicked", () => {
    props.taskDecisions[props.taskDecisions.length - 1].isCallback = true;
    const wrapper = mount(<ContactOutcome {...props} />);

    wrapper.find(".dropdown-icon-btn").at(0).simulate("click");
    wrapper.update();

    wrapper.find(MenuItem).last().simulate("click");
    wrapper.update();

    expect(wrapper.find(CallbackDialog).exists()).toBeTruthy();
    expect(wrapper.props().handleContactOutcomeClick).toHaveBeenCalledTimes(0);
  });

  it("when update callback button is clicked from CallbackDialog, setCallbackDetail method should be emitted", () => {
    const wrapper = mount(<ContactOutcome {...props} />);

    wrapper.find(".dropdown-icon-btn").at(0).simulate("click");
    wrapper.update();

    wrapper.find(MenuItem).last().simulate("click");
    wrapper.update();

    expect(wrapper.find(CallbackDialog).exists()).toBeTruthy();

    Act(() => {
      wrapper
        .find(CallbackDialog)
        .props()
        .setCallbackDetail(new Date(2021, 8, 8), "");
    });
    wrapper.update();

    expect(wrapper.props().setCallbackDetail).toHaveBeenCalled();
  });

  it("when handleCloseDialog method is called from CallbackDialog, should close callback detail dialog", () => {
    const wrapper = mount(<ContactOutcome {...props} />);

    wrapper.find(".dropdown-icon-btn").at(0).simulate("click");
    wrapper.update();

    wrapper.find(MenuItem).last().simulate("click");
    wrapper.update();

    expect(wrapper.find(CallbackDialog).exists()).toBeTruthy();

    Act(() => {
      wrapper.find(CallbackDialog).props().handleCloseDialog();
    });
    wrapper.update();

    expect(wrapper.find(CallbackDialog).exists()).toBeFalsy();
  });
});
