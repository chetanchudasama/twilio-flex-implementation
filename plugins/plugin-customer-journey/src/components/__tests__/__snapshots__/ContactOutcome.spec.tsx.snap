// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactOutcome renders correctly on load 1`] = `
<div
  className="css-5xql2q"
>
  <div
    className="MuiGrid-container-1 MuiGrid-spacing-xs-8-23 MuiGrid-wrap-xs-nowrap-7 MuiGrid-justify-xs-flex-end-19"
  >
    <div
      className="MuiGrid-item-2"
    >
      <button
        className="MuiButtonBase-root-124 MuiButton-root-98 MuiButton-outlined-106 no-contact-btn"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-99"
        >
          No Contact (Left VM)
        </span>
        <span
          className="MuiTouchRipple-root-162"
        />
      </button>
    </div>
    <div
      className="MuiGrid-item-2"
    >
      <button
        className="MuiButtonBase-root-124 MuiButton-root-98 MuiButton-outlined-106 no-contact-btn"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-99"
        >
          No Contact (No VM)
        </span>
        <span
          className="MuiTouchRipple-root-162"
        />
      </button>
    </div>
    <div
      className="MuiGrid-item-2"
    >
      <button
        className="MuiButtonBase-root-124 MuiIconButton-root-127 dropdown-icon-btn"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-132"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-133"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-162"
        />
      </button>
      <li
        className="MuiButtonBase-root-124 MuiListItem-root-145 MuiListItem-default-148 MuiListItem-gutters-153 MuiListItem-button-154 MuiMenuItem-root-142 MuiMenuItem-gutters-143"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="menuitem"
        tabIndex={-1}
      >
        NTU - Customer Not Interested
        <span
          className="MuiTouchRipple-root-162"
        />
      </li>
      <li
        className="MuiButtonBase-root-124 MuiListItem-root-145 MuiListItem-default-148 MuiListItem-gutters-153 MuiListItem-button-154 MuiMenuItem-root-142 MuiMenuItem-gutters-143"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="menuitem"
        tabIndex={-1}
      >
        Warm Transfer to Account Manager
        <span
          className="MuiTouchRipple-root-162"
        />
      </li>
      <li
        className="MuiButtonBase-root-124 MuiListItem-root-145 MuiListItem-default-148 MuiListItem-gutters-153 MuiListItem-button-154 MuiMenuItem-root-142 MuiMenuItem-gutters-143"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="menuitem"
        tabIndex={-1}
      >
        Customer Hung Up During Call
        <span
          className="MuiTouchRipple-root-162"
        />
      </li>
      <li
        className="MuiButtonBase-root-124 MuiListItem-root-145 MuiListItem-default-148 MuiListItem-gutters-153 MuiListItem-button-154 MuiMenuItem-root-142 MuiMenuItem-gutters-143"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="menuitem"
        tabIndex={-1}
      >
        Customer Hung Up During Transfer
        <span
          className="MuiTouchRipple-root-162"
        />
      </li>
      <li
        className="MuiButtonBase-root-124 MuiListItem-root-145 MuiListItem-default-148 MuiListItem-gutters-153 MuiListItem-button-154 MuiMenuItem-root-142 MuiMenuItem-gutters-143"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="menuitem"
        tabIndex={-1}
      >
        Call Did Not Connect
        <span
          className="MuiTouchRipple-root-162"
        />
      </li>
      <li
        className="MuiButtonBase-root-124 MuiListItem-root-145 MuiListItem-default-148 MuiListItem-gutters-153 MuiListItem-button-154 MuiMenuItem-root-142 MuiMenuItem-gutters-143 menu-item-divider"
        onBlur={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="menuitem"
        tabIndex={-1}
      >
        <hr
          className="MuiDivider-root-157 divider"
        />
        <span
          className="MuiTouchRipple-root-162"
        />
      </li>
      <li
        className="MuiButtonBase-root-124 MuiListItem-root-145 MuiListItem-default-148 MuiListItem-gutters-153 MuiListItem-button-154 MuiMenuItem-root-142 MuiMenuItem-gutters-143"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="menuitem"
        tabIndex={-1}
      >
        Arrange a Callback
        <span
          className="MuiTouchRipple-root-162"
        />
      </li>
    </div>
  </div>
</div>
`;
